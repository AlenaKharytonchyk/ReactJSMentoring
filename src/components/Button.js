import React from "react";

export default class Button extends React.Component {
    constructor({buttonName, priority, onClick}) {
        super();
        this.buttonName = buttonName;
        this.priority = priority;
        this.onClick = onClick;
    }

    render() {
        return React.createElement(
            'button',
            {
                className: `button-${this.buttonName}`,
                onClick: () => this.onClick(this.priority),
            },
            this.buttonName);
    }
}
