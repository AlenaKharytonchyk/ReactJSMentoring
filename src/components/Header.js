import React from "react";

export default class TitleElement extends React.Component {
    render() {
        return React.createElement('h1', {className: 'title'}, 'Hello first task');
    }
}
