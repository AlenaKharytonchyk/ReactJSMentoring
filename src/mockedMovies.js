export const moviesArray = [
    {
        image: "https://picsum.photos/323/486?random=1",
        title: "First film",
        description: "Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.—Soumitra",
        rating: "8,2",
        genre: ["drama, comedy"],
        year: "2000",
        duration: "1.4",
        id: 1,
    },
    {
        image: "https://picsum.photos/323/486?random=2",
        title: "Second film with long long title",
        description: "2.Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.—Soumitra",
        rating: "5,2",
        genre: ["drama, comedy, detective"],
        year: "2020",
        duration: "1.4",
        id: 2,
    },
    {
        image: "https://picsum.photos/323/486?random=3",
        title: "Third film",
        description: "3.Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.—Soumitra",
        rating: "9,2",
        genre: ["detective"],
        year: "2010",
        duration: "1.4",
        id: 3,
    },
    {
        image: "",
        title: "Fourth film",
        description: "4.Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.—Soumitra",
        rating: "3,2",
        year: "2010",
        duration: "1.4",
        id: 4,
    }
];
