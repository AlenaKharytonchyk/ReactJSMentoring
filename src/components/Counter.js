import React from "react";
import Button from "./button/Button";
import Title from "./title/Title";
export default class Counter extends React.Component {
    constructor() {
        super();
        this.state={
            initialValue : 0
        }
    }
    counterUpdate (priority) {
        this.setState({initialValue: priority ? this.state.initialValue + 1 : this.state.initialValue - 1 })
    }
    render() {
       return (
           <div>
               <Title />
               <Button
                   buttonName='plus'
                   initialValue={this.state.initialValue}
                   priority={true}
                   onClick={(priority) => this.counterUpdate(priority)}
               />
               { React.createElement('div', {style: {color: '#FFF'}}, this.state.initialValue) }
               <Button
                   buttonName='minus'
                   initialValue={this.state.initialValue}
                   priority={false}
                   onClick={(priority) => this.counterUpdate(priority)}
               />
           </div>
       )
    }
}
