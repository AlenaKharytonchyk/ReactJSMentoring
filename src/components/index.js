import App from "../App";
import Button from "./button/Button";
import Container from "./container/Container";
import GenreSelect from "./genreSelect/GenreSelect";
import InputField from "./input/Input";
import Title from "./title/Title";
import Counter from "./counter/Counter";

export { App, Button, Container, GenreSelect, InputField, Title, Counter };
