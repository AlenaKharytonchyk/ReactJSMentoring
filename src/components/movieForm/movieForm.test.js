import React from "react";
import {MovieForm} from "../../components";
import {render, screen} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import {moviesArray} from "../../mockedMovies";
import {MemoryRouter} from "react-router-dom";

describe("MovieForm", () => {
    it("renders snapshot", () => {
        expect.assertions(1);

        const onClose = jest.fn();

        render(
            <MemoryRouter>
                <div className="App">
                    <MovieForm
                        formTitle="Form title"
                        showModal={true}
                        onClose={onClose}
                        initialMovie={moviesArray[0]}/>
                </div>
            </MemoryRouter>
        );
        const modalComponent = screen.getByTestId('movie-form');
        expect(modalComponent).toMatchSnapshot();
    })

    it("calls onSubmit callback with movie info when click on Submit button", async () => {
        expect.assertions(1);

        const onSubmit = jest.fn();

        const expected = {
            "movie_name": moviesArray[0].title,
            "movie_url": moviesArray[0].poster_path,
            "overview": moviesArray[0].overview,
            "release_date": moviesArray[0].release_date,
            "runtime": moviesArray[0].runtime.toString(),
            "sort": "all",
        }

        render(
            <MemoryRouter>
                <div className="App"><MovieForm
                    formTitle="Form title"
                    submitCallback={onSubmit}
                    showModal={true}
                    initialMovie={moviesArray[0]}/></div>
            </MemoryRouter>
            );

        await userEvent.click(screen.queryByTestId('submit-button'));

        expect(onSubmit).toHaveBeenCalledWith(expected);
    })
})
