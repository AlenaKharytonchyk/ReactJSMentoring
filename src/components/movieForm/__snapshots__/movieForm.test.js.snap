// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieForm renders snapshot 1`] = `
<div
  class="movie-form-container"
  data-testid="movie-form"
>
  <form
    action=""
    method="post"
  >
    <div
      class="field-container"
    >
      <label
        for="title"
      >
        TITLE
      </label>
      <input
        id="title"
        name="movie_name"
        type="text"
        value="Star movie"
      />
    </div>
    <div
      class="field-container"
    >
      <label
        for="release_date"
      >
        RELEASE DATE
      </label>
      <input
        id="release_date"
        name="release_date"
        type="date"
        value="2018-02-07"
      />
    </div>
    <div
      class="field-container"
    >
      <label
        for="url"
      >
        MOVIE URL
      </label>
      <input
        id="url"
        name="movie_url"
        type="text"
        value="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
      />
    </div>
    <div
      class="field-container"
    >
      <label
        for="runtime"
      >
        RUNTIME
      </label>
      <input
        id="runtime"
        name="runtime"
        type="text"
        value="106"
      />
    </div>
    <div
      class="field-container"
    >
      <label
        for="genre"
      >
        GENRE
      </label>
      <select
        data-testid="sorting"
        id="sort"
        name="sort"
      >
        <option
          data-testid="all"
          value="all"
        >
          ALL
        </option>
        <option
          data-testid="comedy"
          value="comedy"
        >
          COMEDY
        </option>
        <option
          data-testid="drama"
          value="drama"
        >
          DRAMA
        </option>
        <option
          data-testid="detective"
          value="detective"
        >
          DETECTIVE
        </option>
      </select>
    </div>
    <div
      class="field-container full-width"
    >
      <label
        for="overview"
      >
        OVERVIEW
        :
      </label>
      <textarea
        id="overview"
        name="overview"
      >
        Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.
      </textarea>
    </div>
    <input
      class="button-transparent button"
      type="reset"
      value="RESET"
    />
    <input
      class="button-pink button"
      data-testid="submit-button"
      type="submit"
      value="SUBMIT"
    />
  </form>
</div>
`;
